src/main/java/org/folio/processing/mapping/mapper/reader/record/marc/MarcRecordReader.java

```diff
@@ -258,11 +258,28 @@ private String retrieveStringFromLastBrackets(String mappingParameter) {
   }
 
   private String retrieveStringWithBracketsFromLastOne(String mappingParameter) {
+    if (mappingParameter.indexOf(FIRST_BRACKET) > mappingParameter.indexOf(SECOND_BRACKET))
+      return mappingParameter.substring(mappingParameter.indexOf(FIRST_BRACKET), mappingParameter.lastIndexOf(SECOND_BRACKET) + 1);
     return mappingParameter.substring(mappingParameter.indexOf(FIRST_BRACKET), mappingParameter.indexOf(SECOND_BRACKET) + 1);
   }
 
   private String retrieveNameWithoutCode(String mappingParameter) {
-    return mappingParameter.substring(0, mappingParameter.trim().indexOf(FIRST_BRACKET) - 1);
+    mappingParameter = mappingParameter.trim();
+    int start = 0, end = mappingParameter.length();
+
+    if (mappingParameter.startsWith(FIRST_BRACKET)) {
+      start++;
+    }
+
+    for (int i = start; i < end; i++) {
```

Comment:
@JavokhirAbdullayev  Just a small point - is there an opportunity to make this mechanism more "readable"? If yes can you improve it a little bit? Thanks!

Reply:
thanks for the suggestion,  increased readability by splitting some functionality to the separate method and some comments